### PR TITLE
Advanced ICL

### PR DESCRIPTION
**Pull request by Nitesh053**
@trigaten  Please Review. Have added Advanced ICL section Under Advanced section

### CODE REVIEW

**Comment by trigaten** [comment]
SyntaxError: /vercel/path0/docs/advanced_applications/ICL.md: Expected corresponding JSX closing tag for . (14:0)
--
10:41:16.251 | 12 \| <p align="center">
10:41:16.251 | 13 \|

**Comment by Nitesh053** [comment]
image

@trigaten  I ran the site locally, Corrected the error.

File: docs/advanced_applications/ICL.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+sidebar_position: 2
+---
+
+# 🟡 Advanced ICL
+
+ICL is an exciting new paradigm in NLP where large language models LLMS make predictions based on contexts augmented with just a few training examples. LLM are able to extract patterns from the examples provided in the context and use them to perform many complex NLP tasks. We'll be discussing recent optimizations of ICL covering methods for fine-tuning LLM for ICL, how to pick good training examples for ICL and how to design great prompts.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:7]: Say in context learning (ICL) and add vocabulary hint (see https://learnprompting.org/contribute#prompt-engineering-techniques).

Comment (trigaten) [inline docs/advanced_applications/ICL.md:7]: LLM are -> LLMs are

File: docs/advanced_applications/ICL.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+## **How Does in-Context Learning Works?**
+
+Here is an illustration of how in-context Learning Works in practice. ICL requires preparing a context that contains a few examples in below figure for the task of sentiment analysis, the examples are arranged to follow a template written in natural language, for example ‘**review** :delicious food’ and ‘**sentiment**: positive’ after the examples a query is inserted in the context, a query is basically an unseen input formatted according to the same template review ‘good meal’ in the figure the full context also called a prompt is presented to a large language model LLM that uses the examples to auto-complete the query resulting in a prediction for the missing sentiment. ICL works not only for simple tasks such as classification but even for complex text generation and reasoning tasks such as summarization, Common Sense reasoning, or even data annotation the power lies in the flexibility of the approach. The same LLM can be used to tackle multiple tasks just by presenting a different context.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:11]: Why capitalize Learning Works?

**Comment by trigaten** [comment]
There needs to be at least one citation (for the paper) in this article.

**Comment by trigaten** [comment]
Properly capitalize glue/squad and cite them

**Comment by trigaten** [comment]
Can you make the prompt design section first-- that is the core of this website.

File: docs/advanced_applications/ICL.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   
+  - Demonstration format
+
+### Demonstration organization
+
+First let's focus on demonstration organization. This involves selecting and then ordering a subset of the training examples you have available for your Target tasks. The selected examples will be used as demonstrations to the LLM. Example selection can be either unsupervised or supervised. 
+
+#### Demonstration Selection
+
+Unsupervised methods for selection include using predefined metrics like L2 distance or cosine similarity distance to select the closest neighbours as demonstrations, or selecting prompts with low perplexity.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:65]: This needs to be explained more

File: docs/advanced_applications/ICL.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+### Demonstration organization
+
+First let's focus on demonstration organization. This involves selecting and then ordering a subset of the training examples you have available for your Target tasks. The selected examples will be used as demonstrations to the LLM. Example selection can be either unsupervised or supervised. 
+
+#### Demonstration Selection
+
+Unsupervised methods for selection include using predefined metrics like L2 distance or cosine similarity distance to select the closest neighbours as demonstrations, or selecting prompts with low perplexity. 
+
+Supervised methods involve using a scoring language model to evaluate the concatenation of each candidate example and the input, labelling high scoring candidates as positive examples and low scoring candidates as hard negative examples. Reinforcement learning has also been used to model the example selection process.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:67]: This stuff absolutely needs to be cited.

File: docs/advanced_applications/ICL.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+####  Instruction Formatting
+
+This involves the design of the prompt itself, including its language and structure. One aspect of formatting is the format of the instructions of the prompt, good instructions are key to good ICL performance, but they can be hard to come by because they rely on human written sentences. LLM can actually help to generate task instructions on their own, given several demonstration examples methods like automatic prompt engineer(APE) use LLM for automatic generation and selection of instructions in summary prompt design is crucial to ICL success and there are many approaches to optimize different aspects of the prompts to boost performance optimizing each of the prompt components is likely to have a big impact on the performance and robustness of ICL.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:79]: This info is covered elsewhere, and not super informative as written.

File: docs/advanced_applications/ICL.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+#### Reasoning Steps Formatting
+
+Adding intermediate reasoning steps between inputs and outputs to construct demonstrations, which are called chain-of-thoughts (CoT). With chain-of-thought prompting, LLMs predict chain-of-thoughts context and then give the final answer. CoT prompting can learn complex reasoning by decomposing input-output mappings into many intermediate steps. As input-output mappings are decomposed into step-by-step reasoning, some researchers apply multi-stage ICL for CoT prompting and design CoT demonstrations for each step. Multi-stage ICL queries LLMs with different prompts in each reasoning step. Self-Ask allows LLMs to generate follow-up questions for the input and ask themselves these questions. Then the questions and intermediate answers will be added to CoTs.
```
Comment (trigaten) [inline docs/advanced_applications/ICL.md:83]: self ask maybe should be a separate page.

**Pull request closed without merging**